test: re-enable upgrade-phonecat-2-hybrid example e2e tests

devversion opened the pull request:
We recently disabled the e2e tests for `upgrade-phonecat-2-hybdrid`
because the switch to APF 10 caused it to fail with an uglify
error. More details in an dedicated issue here: https://github.com/angular/angular/issues/36957.

We can re-enable the test by switching to Terser that properly
handles ES2015 syntax. Previously the rollup bundle consisted
only of ES5 output as the Angular framework packages provided
ES5 output. With APF 10, there is no ES5 output except for the
non-optimizable UMD format.

Hence we either need to downlevel code to ES5, or use Terser
that supports ES2015 syntax.

Closes #36957.

alan-agius4 reviewed (approved):
LGTM

IgorMinar reviewed (approved):
Thank you Paul!

Reviewed-for: global-approvers

gkalpak commented on aio/tools/examples/shared/package.json:
```diff
@@ -80,7 +80,7 @@
     "rollup-plugin-commonjs": "^9.2.1",
     "rollup-plugin-node-resolve": "^4.0.0",
-    "rollup-plugin-uglify": "^1.0.1",
+    "rollup-plugin-terser": "^5.3.0",
```
Please make the same change in the boilerplate `package.json` for SystemJS-based examples: [aio/tools/examples/shared/boilerplate/systemjs/package.json#L680][1]

[1]: https://github.com/angular/angular/blob/18ab1476510257cf77552a66bbdade0ae3be58f9/aio/tools/examples/shared/boilerplate/systemjs/package.json#L68

gkalpak reviewed (approved):
Yay! :tada:

gkalpak commented:
BTW, commit message nits:

- The scope should be `docs-infra`: `test: ...` --> `test(docs-infra): ...`
- Typo: `an dedicated` --> `a dedicated`

devversion committed:
test(docs-infra): re-enable upgrade-phonecat-2-hybrid example e2e tests

We recently disabled the e2e tests for `upgrade-phonecat-2-hybdrid`
because the switch to APF 10 caused it to fail with an uglify
error. More details in a dedicated issue here:

https://github.com/angular/angular/issues/36957.

We can re-enable the test by switching to Terser that properly
handles ES2015 syntax. Previously the rollup bundle consisted
only of ES5 output as the Angular framework packages provided
ES5 output. With APF 10, there is no ES5 output except for the
non-optimizable UMD format.

Hence we either need to downlevel code to ES5, or use Terser
that supports ES2015 syntax.

Closes #36957.
